tools/Azure.Mcp.Tools.Monitor/src/Services/MonitorService.cs

```diff
@@ -353,6 +370,129 @@ public async Task<List<string>> ListTableTypes(
         }
     }
 
+    public async Task<List<ActivityLogEventData>> ListActivityLogs(
+        string subscription,
+        string resourceName,
+        string? resourceGroup = null,
+        string? resourceType = null,
+        double hours = 24.0,
+        ActivityLogEventLevel? eventLevel = null,
+        int top = 10,
+        string? tenant = null,
+        RetryPolicyOptions? retryPolicy = null)
+    {
+        ValidateRequiredParameters(subscription, resourceName);
+
+        if (top < 1)
+        {
+            top = 10;
+        }
+
+        try
+        {
+            // Resolve the resource ID from the resource name
+            var resourceIdentifier = await _resourceResolverService.ResolveResourceIdAsync(
+                subscription, resourceGroup, resourceType, resourceName, tenant, retryPolicy);
+
+            string resourceId = resourceIdentifier.ToString();
+            string subscriptionId = resourceIdentifier.SubscriptionId
+                ?? throw new ArgumentException($"Unable to extract subscription ID from resource ID: {resourceId}");
+
+            // Get the activity logs from the Azure Management API
+            var activityLogs = await CallActivityLogApiAsync(subscriptionId, resourceId, hours, eventLevel, tenant, retryPolicy);
+
+            // Take only the requested number of logs
+            return activityLogs.Take(top).ToList();
+        }
+        catch (Exception ex)
+        {
+            throw new Exception($"Error retrieving activity logs for resource '{resourceName}': {ex.Message}", ex);
+        }
+    }
+
+    private async Task<List<ActivityLogEventData>> CallActivityLogApiAsync(
+        string subscriptionId,
+        string resourceId,
+        double hours,
+        ActivityLogEventLevel? eventLevel,
+        string? tenant,
+        RetryPolicyOptions? retryPolicy)
+    {
+        var returnValue = new List<ActivityLogEventData>();
+
+        string endpoint = string.Format(ActivityLogEndpointFormat, subscriptionId);
+        var uriBuilder = new UriBuilder(endpoint);
+
+        // Build the query parameters
+        string query = $"api-version={ActivityLogApiVersion}";
+
+        // Create the time filter
+        DateTimeOffset startDate = DateTimeOffset.UtcNow.AddHours(-hours).ToUniversalTime();
+        DateTimeOffset endDate = DateTimeOffset.UtcNow;
+        string filter = $"eventTimestamp ge '{startDate:yyyy-MM-ddTHH:mm:ss.fffZ}' " +
+                       $"and eventTimestamp le '{endDate:yyyy-MM-ddTHH:mm:ss.fffZ}' " +
+                       $"and resourceId eq '{resourceId}'";
+
+        if (eventLevel != null)
+        {
+            filter += $" and levels eq '{eventLevel}'";
```

Comment:
Can we wrap the existing filter in a group and then append the next expression? If someone doesn't know this and appends an "or" expression at the end of the previous filter the code may have subtle bug.